src/queue.ts

```diff
@@ -0,0 +1,135 @@
+export function makeQueue<T extends QueueItem>(): Queue<T> {
+  const items: T[] = []
+  const timestampMap: Map<string, number> = new Map()
```

Comment:
Unfortunately we can't use `Map` in any of the plugins or the the libs that are being used them.
It has some unpredictable bugs after being transpiled in the App.